Info-Orbs/src/widgets/webDataElementModel.cpp

```diff
@@ -0,0 +1,316 @@
+#include "widgets/webDataElementModel.h"
+
+String WebDataElementModel::getType() {
+    return m_type;
+}
+
+void WebDataElementModel::setType(String type) {
+    if (m_type != type) {
+        m_type = type;
+
+        if (m_type == "text") {
+            m_typeEnum = TEXT;
```

Comment:
It might make sense to make dedicated `TextWebDataElement` items, (and all other types) so they can read their own properties, and only contain the relevant fields?